grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/redisqueue/RedisTaskStorage.java

```diff
@@ -102,17 +117,24 @@ public TaskState getState(TaskId id) throws GraknBackendException {
     @Override
     public boolean containsTask(TaskId id) {
         try(Jedis jedis = redis.getResource()){
-            String value = jedis.get(id.getValue());
+            String value = jedis.get(encodeKey.apply(id.getValue()));
             return value != null;
         }
     }
 
     @Override
-    public Set<TaskState> getTasks(TaskStatus taskStatus, String taskClassName, String createdBy,
-            EngineID runningOnEngine, int limit, int offset) {
+    public Set<TaskState> getTasks(@Nullable TaskStatus taskStatus, @Nullable String taskClassName,
+            @Nullable String createdBy, @Nullable EngineID runningOnEngine, int limit, int offset) {
         try (Jedis jedis = redis.getResource()) {
-            // TODO change structure of task keys with a prefix. Can we do better than filtering?
-            Stream<TaskState> stream = jedis.keys("*-*-*-*").stream().map(value -> (TaskState) deserialize(Base64.getDecoder().decode(value)));
+            Stream<TaskState> stream = jedis.keys(PREFIX + "*").stream().map(key -> {
```

Comment:
So correct me is I am wrong. Doesn't using a regex in a redis query constitute a complex or slow query? If so do we know how this will scale? Could this become a bottleneck?